### PR TITLE
nixos/doc/md-to-db.sh: handle Docbook inclues in CommonMark

### PR DESCRIPTION
**Pull request by ryantm**
You can do includes like this in markdown to include docbook files

````
```{=docbook}
<xi:include href="rl-2111.section.xml" />
```
````

###### Motivation for this change

The NixOS manual has multiple levels of Docbook includes, so we need some want to temporarily represent this in CommonMark while we transition.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/doc/manual/from_md/development/writing-documentation.chapter.xml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<chapter xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink" xml:id="sec-writing-documentation">
+<chapter xmlns="http://docbook.org/ns/docbook"  xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xi="http://www.w3.org/2001/XInclude" xml:id="sec-writing-documentation">
```
Comment (Mic92) [inline nixos/doc/manual/from_md/development/writing-documentation.chapter.xml:1]: Not sure how this change relates script changes.

File: nixos/doc/manual/md-to-db.sh [hunk 2]
```diff
@@ -24,15 +24,19 @@ mapfile -t MD_FILES < <(find . -type f -regex '.*\.md$')
 for mf in ${MD_FILES[*]}; do
   if [ "${mf: -11}" == ".section.md" ]; then
     mkdir -p $(dirname "$OUT/$mf")
+    OUTFILE="$OUT/${mf%".section.md"}.section.xml"
     pandoc "$mf" "${pandoc_flags[@]}" \
-      -o "$OUT/${mf%".section.md"}.section.xml"
+      -o "$OUTFILE"
+    grep -q -m 1 "xi:include" "$OUTFILE" && sed -i 's|xmlns:xlink="http://www.w3.org/1999/xlink"| xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xi="http://www.w3.org/2001/XInclude"|' "$OUTFILE"
```
Comment (Mic92) [inline nixos/doc/manual/md-to-db.sh:30]: Ah. I see!

**Comment by bobby285271** [comment]
I tested converting `nixos/doc/manual/configuration/profiles.xml` and this is the result https://github.com/bobby285271/nixpkgs/commit/4c38d74dad1ccf8def5aaace413a683f7a901f51, I got something like:

```
&lt;xi:include href=<quote>all-hardware.section.xml</quote> /&gt;
```

Not sure if I am doing something wrong.

File: nixos/doc/manual/from_md/development/writing-documentation.chapter.xml (hunk 1 above)
Comment (ryantm) [reply nixos/doc/manual/from_md/development/writing-documentation.chapter.xml:1]: The from_md xml files need to be updated whenever the script changes, and in this case xi:include is mentioned in the Markdown file, so this changed.

**Comment by ryantm** [comment]
> Not sure if I am doing something wrong.

@bobby285271 Sorry, it is confusing to embed markdown in markdown. You need to use it like this:

````
```{=docbook}
<xi:include href="rl-2111.section.xml" />
```
````

**Comment by bobby285271** [comment]
The build result (and the rendered content) this time looks good (https://github.com/bobby285271/nixpkgs/commit/450886eac02f42f12b42058a82d4fc8dc3e10e08).

**Comment by bobby285271** [comment]
Related: #137082.

**Pull request merged**